src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -346,7 +367,9 @@ def _search_in_extension_commands(self, command_str):
                 cmd_chain = cmd_chain[part]
             except KeyError:
                 return None
-        return None
+        # command_str is prefix of one or more complete commands.
+        all_exts = self._get_all_extensions(cmd_chain)
+        return list(all_exts) if all_exts else None
```

Comment:
if the `command_str` starts with a `not-exist-command`, this function will return all the extension names in root cmd_chain, right?

Reply:
In this case, KeyError will be thrown and return None in above code. This function will not be called.